### DISCUSSION TITLE
how to chomp with memoryview?

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#14088] [Using MicroPython]
I've setup a 20k bytearray and used memoryview to fill 18k of it with bytes from a modem. But now I need to chomp through selectively deleting stuff, starting with the unused to 2k at the end of the bytearray, without running out of ram.
Memoryview will let me selectively alter bytes in the array but I can't seem to remove any with it. If I try slicing out the data I get memory allocation errors. Is there a way to reduce a 20k bytearray down to ~17.7k without using another 17.7k of ram to do it?

**Comment by andrewleech** [comment]
This appears to be a continuation of https://github.com/orgs/micropython/discussions/14080 ? Probably best to keep the discussion in one place.

I don't believe there is a way to just reduce the size of an existing bytearray without a copy being made, (micro)python doesn't really work that way.

**Reply by kjm1102** [reply]
Looks like making a copy is obligatory Andrew. CEFNs comment in https://forum.micropython.org/viewtopic.php?t=4912 had me thinking there might be a work around using memoryview but I can't find it.

**Reply by jimmo** [reply]
@kjm1102 See https://github.com/orgs/micropython/discussions/14090#discussioncomment-8780022

**Reply by andrewleech** [reply]
You can often simply keep the original bytearray untouched and use the memoryview from now on to treat it as a smaller buffer.

If you need to snip out a small section in the middle, joining the start and end sections, this sort of thing probably works:
```
Q[100: 1100] = Q[200: 1200]
```
This would effectively delete the section from 100 to 200, copying the 1000 bytes after 200 down to 100. This should copy the data over itself without needing a separate buffer.
